product_code_unique/__init__.py

```diff
@@ -0,0 +1,12 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import models
+
+
+def pre_init_product_code(cr):
```

Comment:
You will have to treat templates as weel as the other module..

Reply:
Product Templates are not mandatory but required when it has by default 1 variant only. So no need to initialize the value in this case.

Reply:
![image](https://user-images.githubusercontent.com/14207366/39022624-4967dad8-43eb-11e8-8e67-db92426f9c97.png)

Reply:
@feketemihai 

FYI, @agyamuta will change the method to find and remove duplicates default_code in product_product.

We don't need to touch product_template as the default_code of a product template with many variants will be empty:
https://github.com/odoo/odoo/blob/11.0/addons/product/models/product_template.py#L256

The uniqueness constraint is maintained with empty values.